### PR TITLE
Remove pyethereum dependency

### PR DESCRIPTION
**Pull request by palango**
This should be merged after raiden-network/raiden-libs#64 to get clean dependencies.

### CODE REVIEW

File: raiden_contracts/contract_manager.py [hunk 1]
```diff
@@ -54,14 +35,24 @@ def assert_has_solidity(func):
     """decorator - Raise CompileError if HAS_SOLIDITY is set to False"""
     @wraps(func)
     def func_wrap(*args, **kwargs):
-        if _solidity is None:
-            raise _solidity.CompileError(
-                'solc not found in $PATH. Check your path or set $SOLC_BINARY.'
-            )
+        # if _solidity is None:
```
Comment (palango) [inline raiden_contracts/contract_manager.py]: This needs to be replaced.

File: raiden_contracts/contract_manager.py [hunk 2]
```diff
@@ -84,12 +75,10 @@ def __init__(self, path: Union[str, List[str]]) -> None:
     @assert_has_solidity
     def compile_contract(self, contract_name: str, libs=None, *args):
```
Comment (palango) [inline raiden_contracts/contract_manager.py]: Are these functions used anywhere?

**Comment by pcppcp** [comment]
solc update, great! I'll take a look at the CM/compilation test tomorrow.

**Comment by pcppcp** [comment]
I think having a separate venv for `ContractManager` tests is the only option now. Theoretically it should work with current `requirements.txt` file.

**Comment by palango** [comment]
Yeah, next problem is the `compile_contract` function. I'm not able to get unlinked bytecode, so testing won't work.

File: raiden_contracts/contract_manager.py [hunk 3]
```diff
@@ -81,16 +60,19 @@ def __init__(self, path: Union[str, List[str]]) -> None:
...
-            self.get_contract_path(contract_name)[0],
-            contract_name,
-            combined='abi,bin',
-            libraries=libs,
-            extra_args=self.get_mappings()
+        contract_json = compile_files(
+            [self.get_contract_path(contract_name)[0]],
+            output_values=('abi', 'bin'),
+            import_remappings=self.get_mappings(),
+            optimize=True
```
Comment (loredanacirstea) [inline raiden_contracts/contract_manager.py]: Please set to false. We have decided to not use the optimizer for the moment.

File: raiden_contracts/contract_manager.py [hunk 4]
```diff
@@ -125,18 +106,22 @@ def precompile_contracts(contracts_dir: str, map_dirs: List) -> Dict:
...
+        try:
+            res = compile_files(
+                files,
+                output_values=('abi', 'bin'),
+                import_remappings=map_dirs
             )
-        return ret
+            return fix_contract_key_names(res)
+        except FileNotFoundError:
+            raise Exception('Could not compile the contract. Check that solc is avialable.')
```
Comment (loredanacirstea) [inline raiden_contracts/contract_manager.py]: `avialable` -> `available`

**Pull request merged**